pkg/nsx/services/securitypolicy/firewall.go

```diff
@@ -564,115 +576,83 @@ func (service *SecurityPolicyService) createOrUpdateSecurityPolicy(obj *v1alpha1
 	return nil
 }
 
-func (service *SecurityPolicyService) DeleteSecurityPolicy(obj interface{}, isVpcCleanup bool, createdFor string) error {
+func (service *SecurityPolicyService) DeleteSecurityPolicy(obj interface{}, createdFor string) error {
 	var err error
-	switch obj.(type) {
+	switch sp := obj.(type) {
 	case *networkingv1.NetworkPolicy:
-		internalSecurityPolicies, err := service.convertNetworkPolicyToInternalSecurityPolicies(obj.(*networkingv1.NetworkPolicy))
-		if err != nil {
-			return err
+		CRPolicySet := sets.New[string]()
+		CRPolicySet.Insert(service.BuildNetworkPolicyAllowPolicyID(string(sp.UID)))
+		CRPolicySet.Insert(service.BuildNetworkPolicyIsolationPolicyID(string(sp.UID)))
+		for elem := range CRPolicySet {
+			err = service.deleteSecurityPolicy(types.UID(elem), createdFor)
 		}
-		for _, internalSecurityPolicy := range internalSecurityPolicies {
-			err = service.deleteSecurityPolicy(internalSecurityPolicy, isVpcCleanup, createdFor)
-			if err != nil {
-				return err
-			}
-		}
-	case *v1alpha1.SecurityPolicy:
-		err = service.deleteSecurityPolicy(obj, isVpcCleanup, createdFor)
 	case types.UID:
-		err = service.deleteSecurityPolicy(obj, isVpcCleanup, createdFor)
+		err = service.deleteSecurityPolicy(sp, createdFor)
 	}
+
 	return err
 }
 
-func (service *SecurityPolicyService) deleteSecurityPolicy(obj interface{}, isVpcCleanup bool, createdFor string) error {
```

Comment:
Why the flag `isVpcCleanup` is no longer needed?

Reply:
Thanks for reminder, yes, we still need isVpcCleanup flag to instruct the deletion process to follow the same process as vpc mode enabled, Because cleanup process doesn't have vpc enabling flag.